cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_check_manager.py

```diff
@@ -0,0 +1,375 @@
+    if not health_check_conf.health_checks:
+        log.info("No Health Check enabled found")
+
+    return exit_code_sum
+
+
+def _parse_arguments() -> List[str]:
+    parser = argparse.ArgumentParser(description="Health Check Manager")
+    parser.add_argument(
+        "-r",
+        "--region",
+        type=str,
+        required=False,
+        help="Cluster region",
+    )
+    parser.add_argument(
+        "-s",
+        "--stack-name",
+        type=str,
+        required=False,
+        help="Cloudformation stack name",
+    )
+    parser.add_argument(
+        "-n",
+        "--node-type",
+        type=str,
+        required=False,
+        help="Node type",
+    )
+    parser.add_argument(
+        "-q",
+        "--queue-name",
+        type=str,
+        required=False,
+        help="Scheduler queue name",
+    )
+    parser.add_argument(
+        "-cr",
+        "--compute-resource-name",
+        type=str,
+        required=False,
+        help="Scheduler compute resource name",
+    )
+    parser.add_argument(
+        "-j",
+        "--job-id",
+        type=str,
+        required=False,
+        help="Job ID",
+    )
+    parser.add_argument(
+        "-c",
+        "--cluster-configuration",
+        type=str,
+        required=True,
+        help="Path to cluster configuration",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    default_log_file = "/var/log/parallelcluster/slurm_health_check.log"
+    logging.basicConfig(
+        filename=default_log_file,
+        level=logging.INFO,
+        format="%(asctime)s - [%(filename)s:%(funcName)s] - %(levelname)s - JobID %(job_id)s - %(message)s",
+    )
+
+    try:
+        args = _parse_arguments()
+        # Override global log object
+        global log  # pylint: disable=W0603
+        log = logging.LoggerAdapter(log, {"job_id": args.job_id})
+        log.info("HealthCheckManager startup.")
+
+        config_file = os.environ.get("CONFIG_FILE", os.path.join(CONFIG_FILE_DIR, "health_check_manager.conf"))
+        health_check_manager_config = HealthCheckManagerConfig(config_file)
+        try:
+            # Configure root logger
+            fileConfig(health_check_manager_config.logging_config, disable_existing_loggers=False)
+        except Exception as err:
+            if hasattr(err, "message"):
+                err = err.message
+            log.warning(
+                "Unable to configure logging from %s, using default settings and writing to %s.\nException: %s",
+                health_check_manager_config.logging_config,
+                default_log_file,
+                err,
+            )
+        log.info(f"HealthCheckManager config: {health_check_manager_config}")
+        exit_code = _execute_health_checks(health_check_manager_config, args)
+        log.info(f"HealthCheckManager finished with exit code '{exit_code}'.")
+        raise SystemExit(exit_code)
+
+    except Exception as err:
+        if hasattr(err, "message"):
```

Comment:
What about raising/defining another type of exception for "gracefully failures"?

Reply:
I still want to catch all, so to not exit with 1 if an exception isn't managed